m2cgen/interpreters/code_generator.py

```diff
@@ -29,11 +31,32 @@ class BaseCodeGenerator:
 
     def __init__(self, indent=4):
         self._indent = indent
+        self._code_buf = None
         self.reset_state()
 
     def reset_state(self):
         self._current_indent = 0
-        self.code = ""
+        self._finalize_buffer()
+        self._code_buf = StringIO()
+        self._code = None
+        self._finalizer = finalize(self, self._finalize_buffer)
```

Comment:
Great post! https://forum.omz-software.com/topic/4473/python-question-class-cleanup-file-handles-etc/7

Reply:
Nice, yet another Python TIL for me!